### PR TITLE
CXX-498 make OperationException inherit from DBException

### PR DESCRIPTION
**Pull request by amidvidy**
also a bit of cleanup in test exception handling.

### CODE REVIEW

**Comment by amidvidy** [comment]
@TylerBrock, @acmorrow can you review?

File: src/mongo/client/exceptions.cpp [hunk 1]
```diff
@@ -22,7 +22,8 @@ namespace mongo {
     }
 
     OperationException::OperationException(const BSONObj& errorObj)
-        : _lastError(errorObj)
+        : DBException(errorObj.toString(), 0)
+        , _lastError(errorObj)
```
Comment (acmorrow) [inline src/mongo/client/exceptions.cpp]: So.... how many error strings do we have in this exception, now?

**Comment by acmorrow** [comment]
Seems OK to me mod the thing about the strings, feel like we can consolidate?

**Comment by acmorrow** [comment]
LGTM.

**Pull request merged**